pkg/util/util.go

```diff
@@ -33,3 +36,65 @@ func GetRequestHost(req *http.Request) string {
 	}
 	return host
 }
+
+// RefreshValue contains a composite cookie refresh value setting
+type RefreshValue struct {
+	RefreshDuration *time.Duration
+	RefreshPercent  uint8
+}
+
+// Set a value of a refresh
+func (r *RefreshValue) Set(s string) error {
+	origS := s
+	s = strings.TrimSpace(s)
+	if strings.HasSuffix(s, "%") {
+		i, err := strconv.ParseInt(s[:len(s)-1], 10, 8)
+		if err != nil {
+			return fmt.Errorf("percentage \"%s\" must be a valid integer: %s", origS, err)
+		}
+		if i < 1 || i > 100 {
+			return fmt.Errorf("percentage \"%s\" must be an integer between 1 and 100", origS)
+		}
+		r.RefreshPercent = uint8(i)
+	} else {
+		dur, err := time.ParseDuration(s)
+		if err != nil {
+			return fmt.Errorf("\"%s\" must be a valid time duration: %s", origS, err)
+		}
+		if dur < time.Duration(0) {
+			return fmt.Errorf("\"%s\" must be a non-negative time duration", origS)
+		}
+		r.RefreshDuration = &dur
+	}
+	return nil
+}
+
+// IsDurationBased returns true if duration
+func (r *RefreshValue) IsDurationBased() bool {
+	return r.RefreshDuration != nil
+}
+
+// IsDisabled returns true if refresh is disabled
+func (r *RefreshValue) IsDisabled() bool {
+	return r.RefreshDuration != nil && *r.RefreshDuration == time.Duration(0)
```

Comment:
We should make the `time.Duration(0)` a `const` if we do secial logic based on that case because it is the option default.

We did something similar for Flush Interval: https://github.com/oauth2-proxy/oauth2-proxy/blob/aed43a54da328b3f4d85660598818f79c82bcaa3/pkg/apis/options/upstreams.go#L7

Reply:
It wasn't a constant in the previous incarnations and at any rate Duration is type `uint64`, i.e. the compiler will collapse it into a constant anyway, but I'll add it if you like.